eBPF.md

```diff
@@ -132,56 +132,128 @@ Opcode           | Mnemonic | Pseudocode
+0x28            | ldabsh src, dst, imm  | ...
+0x30            | ldabsb src, dst, imm  | ...
+0x38            | ldabsdw src, dst, imm | ...
+0x40            | ldindw src, dst, imm  | ...
+0x48            | ldindh src, dst, imm  | ...
+0x50            | ldindb src, dst, imm  | ...
+0x58            | ldinddw src, dst, imm | ...
+0x61            | ldxw dst, [src+off]   | dst = *(uint32_t *) (src + off)
+0x69            | ldxh dst, [src+off]   | dst = *(uint16_t *) (src + off)
+0x71            | ldxb dst, [src+off]   | dst = *(uint8_t *) (src + off)
+0x79            | ldxdw dst, [src+off]  | dst = *(uint64_t *) (src + off)
+0x62            | stw [dst+off], imm    | *(uint32_t *) (dst + off) = imm
+0x6a            | sth [dst+off], imm    | *(uint16_t *) (dst + off) = imm
+0x72            | stb [dst+off], imm    | *(uint8_t *) (dst + off) = imm
+0x7a            | stdw [dst+off], imm   | *(uint64_t *) (dst + off) = imm
+0x63            | stxw [dst+off], src   | *(uint32_t *) (dst + off) = src
+0x6b            | stxh [dst+off], src   | *(uint16_t *) (dst + off) = src
+0x73            | stxb [dst+off], src   | *(uint8_t *) (dst + off) = src
+0x7b            | stxdw [dst+off], src  | *(uint64_t *) (dst + off) = src
 
 ## Branch Instructions
 
+### 64-bit
+
+Opcode         | Mnemonic            | Pseudocode
+---------------|---------------------|------------------------
+0x05           | ja +off             | PC += off
+0x15           | jeq dst, imm, +off  | PC += off if dst == imm
+0x1d           | jeq dst, src, +off  | PC += off if dst == src
+0x25           | jgt dst, imm, +off  | PC += off if dst > imm
+0x2d           | jgt dst, src, +off  | PC += off if dst > src
+0x35           | jge dst, imm, +off  | PC += off if dst >= imm
+0x3d           | jge dst, src, +off  | PC += off if dst >= src
+0xa5           | jlt dst, imm, +off  | PC += off if dst < imm
+0xad           | jlt dst, src, +off  | PC += off if dst < src
+0xb5           | jle dst, imm, +off  | PC += off if dst <= imm
+0xbd           | jle dst, src, +off  | PC += off if dst <= src
+0x45           | jset dst, imm, +off | PC += off if dst & imm
+0x4d           | jset dst, src, +off | PC += off if dst & src
+0x55           | jne dst, imm, +off  | PC += off if dst != imm
+0x5d           | jne dst, src, +off  | PC += off if dst != src
+0x65           | jsgt dst, imm, +off | PC += off if dst > imm (signed)
+0x6d           | jsgt dst, src, +off | PC += off if dst > src (signed)
+0x75           | jsge dst, imm, +off | PC += off if dst >= imm (signed)
+0x7d           | jsge dst, src, +off | PC += off if dst >= src (signed)
+0xc5           | jslt dst, imm, +off | PC += off if dst < imm (signed)
+0xcd           | jslt dst, src, +off | PC += off if dst < src (signed)
+0xd5           | jsle dst, imm, +off | PC += off if dst <= imm (signed)
+0xdd           | jsle dst, src, +off | PC += off if dst <= src (signed)
+0x85 (src = 0) | call imm            | Helper function call
```

Comment:
Can we get different mnemonics for each of the call variants?

In bpf_conformance I was proposing:
call [modifier] imm

Where [modifier] is optional and can be one of:
helper -> src = 0
local -> src = 1
runtime -> src = 2

Reply:
Thanks for the feedback! I don't mind much either way, but let's see what syntax makes it to the Linux docs if the table gets updated, maybe? This PR has been sitting for nearly four years and I updated mostly to help Dave with the eBPF specs, at this point I doubt it will ever get merged.

Reply:
That's awesome. I didn't realize there was an official set of mnemonics as LLVM and GNU bintools seem to have wildly different disassembly outputs. 

I just want to make sure the bpf_conformance tool has the "correct" ones.

Reply:
I don't think there's anything official, but I know @dthaler intended to have [the table merged to the kernel docs](https://github.com/ebpffoundation/ebpf-docs/pull/1/files#diff-81b91fc4763ca969eb6fbd1394354c341404fef0039d083235788cd042d13a7aR501) (Alexei pushed back initially, at least from having them in the official spec - Not sure where things are right now, I haven't been able to follow over the last weeks). If this happens, it would probably make the document in the current repo irrelevant.